charts/consul/templates/cni-clusterrole.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if (and (.Values.connectInject.cni.enabled) (not .Values.connectInject.transparentProxy.defaultEnabled)) }}{{ fail "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" }}{{ end -}}
```

Comment:
Why do we require tproxy to be default enabled? I don't think it needs to be since CNI could be installed with tproxy default disabled and read tproxy annotations on the pod or the namespace.

Reply:
We should gate it on or fail if connectInject is disabled.

Reply:
Ok, this was my miss-understanding then since CNI is a continuation of tproxy. I will update this and a bunch of bats tests.

Reply:
I have updated all of the templates to check for connectInject.cni.enabled + connectInject.enabled, updated all the bats tests and removed that imageK8s check.

Thanks for looking at this and catching these things!